classes/requests/checkout/class-paysoncheckout-for-woocommerce-update-reference.php

```diff
@@ -18,25 +18,33 @@ class PaysonCheckout_For_WooCommerce_Update_Reference extends PaysonCheckout_For
 	 *
 	 * @param string|null $order_id The WooCommerce order id.
 	 * @param array       $payson_data The Payson order data.
-	 * @return array
+	 * @return array|WP_Error|false If the request could not be issued, FALSE is returned. Otherwise, WP_Error or an array.
 	 */
 	public function request( $order_id = null, $payson_data = null ) {
-		$payment_id   = null !== WC()->session && ! empty( WC()->session->get( 'payson_payment_id' ) ) ? WC()->session->get( 'payson_payment_id' ) : $payson_data['id'];
+		$payment_id   = WC()->session->get( 'payson_payment_id', null ) ?? WC()->session->get( 'payson_payment_id' ) ?? $payson_data['id'] ?? null;
```

Comment:
I dont believe this line will work for what you are intending.

Since `WC()` returns an instance of the `WooCommerce` class, and this class does have the property `session` already set to `null` as default, you will need to check against this before you use the `get()` method.

If you change it to this though:
`WC()->session?->get( 'payson_payment_id') ?? $payson_data['id'] ?? null;`

It should do the same as what you want at the moment. `?` after session to check for null value, and then skip the default value in get since its already that by standard in WC.

Reply:
I've made changes accordingly.